---
title: "Deseq2 and heatmap "
author: "Yang Lyu"
date: "`r Sys.Date()`"
output:
  pdf_document:
    keep_tex: true
    latex_engine: pdflatex
    fig_caption: yes
    highlight: haddock
    number_sections: yes
    #latex_engine: xelatex

knit: (
  function(inputFile, encoding) { 
      pSubTitle <- paste(Sys.Date(), "JH paper RNAseq data" )
     
    rmarkdown::render( 
      input       = inputFile, 
      encoding    = encoding, 
      params      = list(sub_title = pSubTitle),      
      output_file = pSubTitle,
      output_dir="/Volumes/lyu.yang/MAC_1/R/Project/11_JH_biobank_paper/Output/RNAseq/Github" ) 
       }
       
       )
      
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo =F,message = F,warning = F,include = F)
```


# RNAseq data loading
```{r,include=F}
getwd()
rm(list=ls())
#dir.create("/Volumes/lyu.yang/MAC_1/R/Project/10_JH-2002-shURB5/Output")
Path="/Volumes/lyu.yang/MAC_1/R/Project/11_JH_biobank_paper"
Rcode_path=file.path(Path,"Rcode")
Rdata_path=file.path(Path,"Rdata")
Rawdata_path=file.path(Path,"Rawdata")
Reference_path="/Volumes/lyu.yang/MAC_1/R/Public_database/Reference"
```


```{r,include=F}
library(dplyr)
library(ggplot2)
 
```

```{r}
# R_function_path="/Volumes/lyu.yang/MAC_1/R/Public_database/R_function"
# Reference_path="/Volumes/lyu.yang/MAC_1/R/Public_database/Reference"
# 
# source( file.path(R_function_path, "RNAseq_pipeline.R"))
```

```{r}
# All_sample_anno=readxl::read_excel("/Volumes/lyu.yang/MAC_1/R/Project/NTAP/Rawdata/Sample_annotation/Round4/Round1_4_annotation_03262023.xlsx") 
# 
# JH_paper_anno_RNAseq=All_sample_anno %>%dplyr::filter(batch!="WU_batch3") %>%dplyr::filter(grepl("RNA",Assay))%>%dplyr::filter(grepl("tumor|blood", tissue))%>%mutate(RNA_ID2=gsub("\\.","-",RNA_ID))

```

```{r}

# GSEA_pathway_list in the reference
load(file.path(Reference_path,"Human_GSEA","GO_KEGGP_WP_C2CGP_Reactome_pathway_list.rdata"))
# Function
PCA_Plot_3<- function (data,Annotation,VAR,Color) {
  # logcountdata row:genes,column: samples
  pca <- prcomp(data) 
  pca_out<-as.data.frame(pca$x)
  df_out<- pca_out %>%tibble::rownames_to_column(var=VAR)  %>% left_join(., Annotation) 
  #df_out<- merge (pca_out,Annotation,by.x=0,by.y=0) 
  
  # label_color<- factor(df_out[,group])
  ggplot(df_out,aes_string(x="PC1",y="PC2")) +geom_point(aes_string(colour = Color))
}

Batch_count_plot=function(logcount,sample_anno){
  
  logcount_2=logcount%>%t() %>%as.data.frame() %>%mutate(File_name=rownames(.))
  Level=sample_anno %>%arrange(batch)%>%.$File_name
  data_mod = reshape2::melt(logcount_2, id.vars='File_name', 
                            measure.vars=rownames(logcount)) %>%left_join(sample_anno)%>%arrange(batch)
  data_mod$File_name=factor(data_mod$File_name,levels =Level )
  #return(logcount_2)
 ggplot(data_mod)+geom_boxplot(aes(x=File_name, y=value, color=batch))+xlab("Sample")+ylab("log2(Count+1)")+theme(axis.text.x=element_blank())
  
  
}

Deseq2_Deseq_function_2<- function (Countdata,Coldata) {
  dds_fil <- DESeq2:: DESeqDataSetFromMatrix(countData =Countdata, colData = Coldata,
                                             design = ~Group)

  # keep <- rowSums(DESeq2::counts(dds_fil)) >= 10
  # dds_fil_2  <- dds_fil[keep,]
  dds_fil_Deg<- DESeq2::DESeq(dds_fil)
  return(dds_fil_Deg)
  "Due to the large number of tests performed in the analysis of RNA-seq and other genome-wide experiments, 
  the multiple testing problem needs to be addressed. A popular objective is control or estimation of the false discovery rate (FDR). 
  Multiple testing adjustment tends to be associated with a loss of power,
  in the sense that the false discovery rate for a set of genes is often higher than the individual p-values of these genes.
  However, the loss can be reduced if genes are omitted from the testing that have little or no chance of being detected as differentially expressed, 
  provided that the criterion for omission is independent of the test statistic under the null hypothesis [21] (see Methods). 
  DESeq2 uses the average expression strength of each gene, across all samples, as its filter criterion, 
  and it omits all genes with mean normalized counts below a filtering threshold from multiple testing adjustment. D
  ESeq2 by default will choose a threshold that maximizes the number of genes found at a user-specified target FDR. 
  In Figures 2A-B and 3, genes found in this way to be significant at an estimated FDR of 10% are depicted in red. "
  
}

Deseq_vst=function (Countdata,Coldata) {
  deseq2Data <- DESeq2:: DESeqDataSetFromMatrix(countData =Countdata, colData = Coldata,
                                             design = ~Group)
  deseq2VST <- DESeq2::vst(deseq2Data)
  
  # Convert the DESeq transformed object to a data frame

  deseq2VST <-   SummarizedExperiment::assay(deseq2VST)
  #deseq2VST <- as.data.frame(deseq2VST)
   
  head(deseq2VST)
  return(deseq2VST)
}

Deseq_norm_counts=function (Countdata,Coldata) {
  deseq2Data <- DESeq2:: DESeqDataSetFromMatrix(countData =Countdata, colData = Coldata,
                                                design = ~Group)
  dds <- DESeq2::estimateSizeFactors(deseq2Data)
  
  # Convert the DESeq transformed object to a data frame
  

  normalized_counts <- DESeq2::counts(dds, normalized=TRUE) %>%as.data.frame()
}

Pathway_enrichment_analysis=function(singscore_name,cutoff) {
  #Data=data.frame(Name=singscore_name) %>%mutate(Split=unlist(strsplit(Name,split="_")))
  split_list=list()
  for (i in 1:length(singscore_name)) {
    Split=unlist(strsplit(singscore_name[i],split="_"))
    split_list[[singscore_name[i]]]=Split
  }
  
  combine=unlist(split_list) %>%gsub("TP53","P53",.)
  common_words=c(c("UP","DN","GOBP","VS","OF","CELLS","CELL","TARGETS","FETAL","DESCARTES","GOMF","ACTIVITY","REACTOME","HP","MODULE","GOCC","RESPONSE","TRANSPORT","REGULATION","COMPLEX","UP.V1","KO","ABNORMAL","IN","PROCESS","CTRL","PBMC","ION","SIGNALING","WITH","BY","WP","KEGG","DN.V1","WT","A","B","C","D","E","PROTEIN","GENES","POS","HALLMARK","VIA","THE","AND","TO","CANCER","TUMOR","PATHWAY","POSITIVE","TREATED","BINDING"),c(1:9))
  
  combine_stat=data.frame(split=combine)%>%subset(!split%in%common_words)%>%  group_by(split) %>% summarise(n=n()) %>% dplyr::filter(!grepl("DAY",split )) %>% arrange(desc(n)) %>%dplyr::filter(n>=cutoff)
  

  
}

Pathway_select_by_keyword=function(singscore_name,keyword){
  Pathway_list=list()
  for (i in 1:length(singscore_name)) {
    Split=unlist(strsplit(singscore_name[i],split="_"))
    Intersect=intersect(Split,keyword)
    if(length(Intersect)>0){
      Pathway_list[[ singscore_name[i]]]=Intersect[1]
      
    }
    
  }
  
  return(Pathway_list)
  }

Singscore_data_plot_2=function(gene_list,expression_data,expression_data_anno,Annotation_File,stat_method){
  Color=c("blue","red")
  Gene_singscore= singcore_Normalized_GSEA(expression_data,gene_list)
  
  Annotation_file=Annotation_File[,c("Group")]  %>%as.data.frame()
  rownames(Annotation_file)=Annotation_File$Sample_name
  p2=GGplot_singscore_boxplot_2(Gene_singscore[[1]],Annotation_File,"Group",Color,"Sample_name",Method=stat_method)
  
  pheatmap_pathway_3 (expression_data_anno,unlist(gene_list),names(gene_list), Annotation_file,"V1")
  list(p2)
}


singcore_GSEA=function (data,genelist, is.na=T) {
  nm=names(genelist)
  plist=list()
  Data_prepare=singscore::rankGenes (data)
  
  for (i in seq_along(genelist)) { 
    scoredf<- singscore::simpleScore (Data_prepare, upSet=unique(genelist[,i]))%>%.[1]
    colnames(scoredf)=paste0(nm[i],"_singscore")
    plist[[i]]=scoredf
  }
  return(plist)
  #output=do.call("cbind",plist)
}

singcore_Normalized_GSEA=function (data,genelist, is.na=T) {
  nm=names(genelist)
  plist=list()
  Data_prepare=singscore::rankGenes (data)
 # n_genes=length(intersect(genelist,rownames(data)))
  
   if( is.list(genelist)) {
    for (i in 1:length(genelist)) { 
      n_genes=length(intersect(unique(genelist[[i]]),rownames(data)))
      if (n_genes>4){
      scoredf<- singscore::simpleScore (Data_prepare, upSet=unique(genelist[[i]]))%>%.[1]
      scoredf$Normalized=scale(scoredf[,1])
      colnames(scoredf)=c(paste0(nm[i],"_singscore") , paste0(nm[i],"_Normalized_singscore" ))
      if(nrow(scoredf)==ncol(data)) {
        plist[[nm[i]]]=scoredf
      }
      }
     }
  }
  if(is.data.frame(genelist))  {
    
  for (i in seq_along(genelist)) { 
    n_genes=length(intersect(unique(genelist[[i]]),rownames(data)))
    if (n_genes>4){
    scoredf<- singscore::simpleScore (Data_prepare, upSet=unique(genelist[,i]))%>%.[1]
    scoredf$Normalized=scale(scoredf[,1])
    colnames(scoredf)=c(paste0(nm[i],"_singscore") , paste0(nm[i],"_Normalized_singscore" ))
    if(nrow(scoredf)==ncol(data)) {
      plist[[nm[i]]]=scoredf
     }
    }
  }
  }
  return(plist)
  #output=do.call("cbind",plist)
}

GGplot_singscore_boxplot_2=function(singscore,Annotation,group_col,color_set,Var,Method) {
  library(ggplot2)
  library(ggpubr)
  gene_name=colnames(singscore)[2]
  Data=singscore%>%tibble::rownames_to_column(.,var=Var)
  Data2= Data %>% left_join(.,Annotation,by=Var)%>%as.data.frame()
  
  
  ggplot(Data2,aes_string(x="Group",y=gene_name,fill=group_col)) +geom_boxplot() + theme(axis.text.x = element_text(angle=90, vjust=.5, hjust=1))+
    ylab(gene_name)+scale_fill_manual(name = group_col, values = color_set) +stat_compare_means(method = Method)
  
  
  
}

pheatmap_pathway_3<- function (expression_data, Gene_select,pathway_name,Annotation,Label_def) {
 # Gene_select<-Pathway_list[,pathway_name]%>%unique()
  Gene_select_anno= expression_data[,colnames(expression_data) %in% Gene_select] %>%t()
 
  # Anno_expression_data=Gene_select_anno[,c("SYMBOL",Group_select)] %>% as.data.frame() %>% distinct() %>% na.omit()
  # rownames(Anno_expression_data)=Anno_expression_data[,"SYMBOL"]
  # Annotation=group_anno["Gene_type"]
  # input= Anno_expression_data[,Group_select]
  # F2_pheatmap <- pheatmap::pheatmap(input, cellwigermline calling GATKdth = 10, cellheight = 12, scale = "row",
  #                                   treeheight_row = 5,
  #                                   show_rownames = T,show_colnames = T, 
  #                                   annotation_col= Annotation,
  #                                   # annotation_row=Annotation,
  #                                   annotation_legend=Label_def,
  #                                   cluster_rows = T,  cluster_cols = F,clustering_distance_rows = "euclidean")
  pheatmap::pheatmap(Gene_select_anno, cellwigermline= 2, cellheight =8, scale = "row",
                     treeheight_row = 5,main=pathway_name,
                     show_rownames = T,show_colnames = F,
                     annotation_col= Annotation,
                     # annotation_row=Annotation,
                     #annotation_legend=Label_def,
                     cluster_rows = T,  cluster_cols = F,clustering_distance_rows = "euclidean")
  
}


Valcano_plot_rnaseq=function(Data,Pathway_list,pathway_name){
  pathway_genes=Pathway_list [[pathway_name]]%>%as.character()
  
  library(ggrepel)
  Data=Data%>%.[.$gene_symbol%in%pathway_genes,]
  # add a column of NAs
  de=Data %>%mutate(diffexpressed=ifelse(padj>0.05|is.na(padj),"NO",ifelse(log2FoldChange>0,"UP","DOWN")))

  de$delabel <- NA
 
  de$delabel[de$diffexpressed != "NO"] <- de$gene_symbol[de$diffexpressed != "NO"]
   # plot adding up all layers we have seen so far
  ggplot(data=de, aes(x=log2FoldChange, y=-log10(padj), col=diffexpressed, label=delabel)) +
    geom_point() + ggtitle(pathway_name)+
    theme_minimal() +
    geom_text_repel() +
    scale_color_manual(values=c("DOWN"="blue", "NO"= "black","UP"= "red")) +
    geom_vline(xintercept=c(-0.6, 0.6), col="red") +
    geom_hline(yintercept=-log10(0.05), col="red")
  
  
  
  
}


Bar_plot_RNAseq_4=function(data,Res,pathway_name,x_name,fill_name,compare_list,data_type) {
  library(ggpubr)
  my_comparisons <- compare_list
  stat.test=Res[rownames(Res) %in% pathway_name,]
  
  Data=data[,c(pathway_name,"Group")] %>%setNames(c("Gene","Group"))
 
  yposition=max(Data$Gene)
  stat.test=Res[rownames(Res) %in% pathway_name,]  %>% mutate(y.position = 1,".y."="Gene")%>%
    mutate("p.format"=ifelse(padj<0.001,"<0.001",ifelse(padj<0.01,"<0.01",ifelse(padj<0.05,"<0.05",round(padj,2)))) ,method="Deseq2")  %>%
    rename(p=pvalue,"p.adj"="padj",)   %>% mutate(y.position = yposition+0.5 ) 
  # return(stat.test)
  # stat.test[,1]$.y.=pathway_name
  
  
  #  stat.test <- compare_means(
  #    Gene ~ Group, data = Data,
  #    method = "t.test"
  #  )
  #  
  #  
  #  
  # return(stat.test)
  p=ggplot(data,aes_string(
    x =x_name,
    y = pathway_name,
    fill=fill_name
  ) )+  geom_violin( ) +
    geom_boxplot(width=0.1) + theme(plot.title=element_text(size=5),legend.position="none")+
    
    scale_fill_brewer(palette="Dark2")+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 
  
  p=ggboxplot(Data, x = "Group", y = "Gene",fill="Group")+ theme(plot.title=element_text(size=5),legend.position="none")+ylab(paste0(pathway_name,"_",data_type))+xlab("")+
    scale_fill_brewer(palette="Dark2")+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 
  p + stat_pvalue_manual(stat.test,label = "p.format")
  
}

Batch_boxplot_4=function(pathway_list,Res,data,x_name,fillby,nCol,Compare_list,data_type) {
  library(gridExtra)
  
  plist=list()
  for(i in 1:length(pathway_list)) {
    Pathway=pathway_list[i]
    # return(Pathway)
    plist[[i]]=Bar_plot_RNAseq_4(data,Res,Pathway,x_name,fillby,Compare_list,data_type )
    
  }
  #plist[["legend"]]=UMAP_legend_2(data, pathway_list[1] ,fillby,fillby)
  
  do.call("grid.arrange", c(plist, ncol=nCol))
  
}

Heatmap_simple=function(gene_list,expression_data,expression_data_anno,Annotation_File){
   
  Annotation_file=Annotation_File[,c("Group")]  %>%as.data.frame()
  rownames(Annotation_file)=Annotation_File$Sample_name
  pheatmap_pathway_3_no_genename (expression_data_anno,unlist(gene_list),names(gene_list), Annotation_file,"V1")
  #list(p2)
}

pheatmap_pathway_3_no_genename<- function (expression_data, Gene_select,pathway_name,Annotation,Label_def) {
  Gene_select_anno= expression_data[,colnames(expression_data) %in% Gene_select] %>%t()
  pheatmap::pheatmap(Gene_select_anno, cellwigermline= 2, cellheight =4, scale = "row",cellwidth = 4,
                     treeheight_row = 5,main=pathway_name,
                     show_rownames = F,show_colnames = F,
                     annotation_col= Annotation,
                     # annotation_row=Annotation,
                     #annotation_legend=Label_def,
                     cluster_rows = T,  cluster_cols = F,clustering_distance_rows = "euclidean")
  
  
}
```

```{r}
dir(Rawdata_path)
```


```{r}
Rawcounts=read.csv(file.path(Rawdata_path,"RNAseq","52_JH_Tumor_raw_counts.csv"),row.names = 1,check.names = F)
Protein_coding_counts=read.csv(file.path(Rawdata_path,"RNAseq","52_JH_Tumor_proteincoding_salmon_counts.csv"),row.names = 1,check.names = F)
RNAseq_anno=read.csv(file.path(Rawdata_path, "RNAseq","52_JH_Tumor_RNAseq_annotation.csv"))
RNAseq_anno$batch=factor(RNAseq_anno$batch,c("JH_batch1" ,"WU_batch1","WU_batch2","WU_PDX_batch1","WU_PDX_batch2"))

Disease_sum=RNAseq_anno%>%group_by(tumorType)%>%summarise(n=n())

setdiff(RNAseq_anno$parentBiospecimenID,colnames(Protein_coding_counts))
```

# Batch effect 
## Raw data batch effect
```{r}
Tumor_count_log=log2(Protein_coding_counts+1)

PCA_Plot_3  (t(Tumor_count_log),RNAseq_anno,"File_name","batch")

Batch_count_plot(Tumor_count_log,RNAseq_anno)

```



## Batch effect was removed by normalization
 
```{r,eval=T}

batches = sapply(as.character(RNAseq_anno$batch), switch, "WU_PDX_batch1"=4,"WU_PDX_batch2"=5, "WU_batch1" = 2, "WU_batch2" = 3,"JH_batch1"=1,USE.NAMES = F)
 
corrected_data = sva::ComBat(Tumor_count_log,batch =RNAseq_anno$batch,par.prior=T, mean.only = F)
 
```


```{r,eval=T}
PCA_Plot_3  (t(corrected_data),RNAseq_anno,"File_name","batch")
#PCA_Plot_3  (t(corrected_data),RNAseq_anno,"File_name","tissueType")
PCA_Plot_3  (t(corrected_data),RNAseq_anno,"File_name","tumorType")
```

```{r}
Batch_count_plot(corrected_data,RNAseq_anno)
```

# 14 MPNST vs 30 PN (normalized counts, DEseq2)


```{r}
GTAC_Sample_anno=RNAseq_anno %>%mutate(Group=Disease) %>%mutate(Sample_name=File_name)
GTAC_Sample_anno$Group=factor(GTAC_Sample_anno$Group,levels=c("PN","cNF","DifNF","NodNF","MPNST"  ))
rownames(GTAC_Sample_anno)=GTAC_Sample_anno$Sample_name
colnames(corrected_data)==GTAC_Sample_anno$Sample_name
corrected_data_counts=2^(corrected_data) %>%round(.,0)
MPNST_vs_NF_Deseq=Deseq2_Deseq_function_2(corrected_data_counts,GTAC_Sample_anno)
DESeq2::resultsNames(MPNST_vs_NF_Deseq)
#"Group_MPNST_vs_PN"
MPNST_vs_PN_res <- DESeq2::results(MPNST_vs_NF_Deseq, name = "Group_MPNST_vs_PN", alpha = 0.05) %>%as.data.frame()
MPNST_vs_PN_res_q005=MPNST_vs_PN_res %>%dplyr::filter(padj<0.05)
MPNST_vs_PN_res_q005_up=MPNST_vs_PN_res_q005 %>%dplyr::filter(log2FoldChange>0)%>%arrange(desc(log2FoldChange))

MPNST_vs_PN_res_q005_DN=MPNST_vs_PN_res_q005 %>%dplyr::filter(log2FoldChange<0)%>%arrange(log2FoldChange)
#
Expression_vst=Deseq_vst(corrected_data_counts,GTAC_Sample_anno)
Expression_vst_anno=t(Expression_vst) %>%as.data.frame() %>%tibble::rownames_to_column (.,var="Sample_name" ) %>%left_join(.,GTAC_Sample_anno)%>% arrange(Group,Sample_name)
rownames(Expression_vst_anno)=Expression_vst_anno$Sample_name

Expression_vst_anno_MPNST_PN=Expression_vst_anno%>%.[grepl("PN|MPNST",.$Sample_name) ,]
Expression_vst_MPNST_PN=Expression_vst[,rownames(Expression_vst_anno_MPNST_PN)]
MPNST_vs_PN_res_2=MPNST_vs_PN_res%>%mutate(gene_symbol=rownames(.))

```

```{r}
MPNST_PN_sample=data.frame(File_name=colnames(Expression_vst_MPNST_PN))%>%left_join(.,GTAC_Sample_anno[,c("File_name","individualID","tumorType")])
knitr::kable(MPNST_PN_sample,caption = "PN and MPNST sample list")%>% kableExtra::kable_styling(latex_options = c("scale_down" ))
```

## Top 20 exprssed genes
```{r}
q005_gene=rownames(MPNST_vs_PN_res_q005)
Expression_vst_anno_MPNST_PN_q005=Expression_vst_anno_MPNST_PN[,c("Sample_name",q005_gene)]
Expression_vst_MPNST_PN_q005=Expression_vst_MPNST_PN[q005_gene,]

Top20_up_genes=MPNST_vs_PN_res_q005%>%arrange(desc(log2FoldChange))%>%.[1:20,]
Down20_up_genes=MPNST_vs_PN_res_q005%>%arrange(log2FoldChange)%>%.[1:20,]

knitr::kable(Top20_up_genes[,c(2,5,6)],caption = "Top 20 Upregulated genes (MPNST vs PN) " ) %>%kableExtra::kable_styling()
knitr::kable(Down20_up_genes[,c(2,5,6)],caption = "Top 20 Downregulated genes (MPNST vs PN) " ) %>%kableExtra::kable_styling(latex_options = c("hold_position"))

```


# GSEA analysis

```{r}

res_q005_rank=MPNST_vs_PN_res_q005%>%arrange(log2FoldChange) %>%.[c("log2FoldChange")]%>%as.data.frame() %>%tibble::rownames_to_column()
GMT_files=list.files(file.path(Reference_path,"Human_GSEA"),pattern="all.v7.5.1.symbols.gmt.txt",full.names = T)
res_q005_rank=MPNST_vs_PN_res_q005%>%arrange(log2FoldChange) %>%.[c("log2FoldChange")]%>%as.data.frame() %>%tibble::rownames_to_column()
ranks <- tibble::deframe(res_q005_rank)
fgsea_list=list()
for (i in GMT_files){
 name=gsub(".*/","",i)
 name2=gsub("all.v7.5.1.symbols.gmt.txt","",name)
 pathway= fgsea::gmtPathways(i)
 fgseaRes <- fgsea::fgseaMultilevel(pathway, ranks,
                           minSize=10,
                           maxSize=500
                           )
 fgsea_list[[name2]]=fgseaRes

}
```

```{r}
All_pathway_singscore=do.call("rbind", fgsea_list)%>%arrange(desc(NES))
# cgwtools::resave(All_pathway_singscore,file=file.path(Rawdata_path,"RNAseq","MPNST_VS_PN_res.rdata"))
All_pathway_singscore_q005=do.call("rbind", fgsea_list)%>%dplyr::filter(padj<0.05)

Hallmark=All_pathway_singscore %>%dplyr::filter(grepl("HALLMARK",pathway )) %>%  arrange(padj)
Hallmark_q005=Hallmark%>%dplyr::filter(padj<0.05)%>% arrange(desc(NES))

knitr::kable(Hallmark_q005[,c(1,2,3,6)],caption = "Hallmark pathways changed in MPNST groups (vs PN,padj<0.05)") %>%kableExtra::kable_styling(latex_options = c("hold_position"))

```



```{r}

key_words_enrichment=Pathway_enrichment_analysis(All_pathway_singscore_q005$pathway,5)
Int_key_words=c("H3K27ME3","P53","TP53","IL4","IL12","TGFB","TNF","MESENCHYMAL","APOPTOSIS","CYCLE","METASTASIS","MITOTIC","STROMAL","EPITHELIAL","MESENCHYMAL","AKT","PI3K","PATHWAY","MYC","EGFR","ERBB","WNT","JNK","P38","RHO","ARF","PDGF","GROWTH_FACTOR","MHC")
Key_pathway=do.call("rbind", Pathway_select_by_keyword(All_pathway_singscore_q005$pathway,Int_key_words) ) %>%as.data.frame() %>%setNames("Keyword") %>%
       mutate(Keyword=gsub("TP53","P53" ,Keyword)) %>%arrange(Keyword)


Key_pathway_anno=All_pathway_singscore_q005 %>%.[.$pathway%in% rownames(Key_pathway) ,] %>%.[,c("pathway","padj","NES")]
```



# Significant pathways
```{r,include=T}
Sig_pathways=c("MEISSNER_BRAIN_HCP_WITH_H3K27ME3","FISCHER_G2_M_CELL_CYCLE" ,"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" ,"WINNEPENNINCKX_MELANOMA_METASTASIS_UP")
knitr::kable(Key_pathway_anno[Key_pathway_anno$pathway%in%Sig_pathways,],caption = "Significant pathways in MPNST (vs PN) " ) %>%kableExtra::kable_styling(latex_options = c("hold_position"))
```

\newpage
## MPNST markers

**S100B, CNP,PMP22,ngfr ,Expression of Schwann cell differentiation markers Downregulaed in MPNST** \t
**TWIST1,SOX9,SOX10  ,stem cell marker, upregulated in MPNST"**
**E-cadherin (CDH1) was downregulated in both neurofibroma and MPNSTs**
**GAS1 (Growth Arrest Specific 1) is a Protein Coding gene. IFG2, PTK7, FGFR1, TWIST1, GAS1, and EGFR.upregulated in MPNST**

```{r,include=T}
MPNST_markers= c( "UBR5","TP53", "CNP","PMP22","NGFR","SOX9","SOX10","S100B","S100A4", "S100A1", "S100A2", "S100A6","CCNB2","TWIST1","EGFR","FGFR1","ERBB3","ERBB2","COL6A3","CDH1","CDKN2A","PDGFA","PDGFB","TGFBR2","GAS1","TGFB1","TGFB2","IGF2","PTK7")

Pathway_list=list("MPNST_markers"=MPNST_markers)

Stat_method="wilcox.test"
gene_list2=data.frame(Pathway=MPNST_markers)
colnames(Expression_vst_MPNST_PN_q005)==MPNST_PN_sample$File_name

Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)


```

```{r,include=T}
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,Pathway_list,"MPNST_markers")
```
 

```{r,include=F,fig.height=35,fig.width=18,out.height="95%"}
Compare_list=list(c("MPNST","PN"))
setdiff(MPNST_markers,colnames(Expression_vst_anno_MPNST_PN))

res_2=MPNST_vs_PN_res%>%mutate(group1="MPNST",group2="PN")
Batch_boxplot_4(MPNST_markers,res_2,Expression_vst_anno_MPNST_PN,x_name="Group",fillby="Group",nCol=4,Compare_list=Compare_list,"Normalized_counts")
```

\newpage

## MEISSNER BRAIN HCP WITH H3K27ME3

```{r,fig.height=8,fig.width=6,out.height="80%"}
Pathway="MEISSNER_BRAIN_HCP_WITH_H3K27ME3"
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Heatmap_simple(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno)

```


```{r,fig.height=16,fig.width=10,out.height="80%",include=T}
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)

```
\newpage
```{r,include=T}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)

```

## FISCHER G2 M CELL CYCLE

```{r,fig.height=8,fig.width=6,include=F}
Pathway="FISCHER_G2_M_CELL_CYCLE" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Heatmap_simple(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno)

```


```{r,fig.height=12,fig.width=8,include=T}
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)
```
\newpage

```{r,include=T}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)

```

## HALLMARK EPITHELIAL MESENCHYMAL TRANSITION

```{r,fig.height=8,fig.width=6}
Pathway="HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Heatmap_simple(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno)

```


```{r,fig.height=12,fig.width=8,include=T}
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)
```

\newpage

```{r,include=T}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)

```

## WINNEPENNINCKX MELANOMA METASTASIS UP


```{r,fig.height=8,fig.width=6}
Pathway="WINNEPENNINCKX_MELANOMA_METASTASIS_UP" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Heatmap_simple(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno)

```


```{r,fig.height=12,fig.width=8}
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)

```


```{r}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)

```


\newpage

## MAPK/AKT PATHWAY

### WP_PI3KAKT_SIGNALING_PATHWAY
```{r,fig.height=8,fig.width=6}
MAPK_AKT=All_pathway_singscore%>%dplyr::filter(grepl("MAPK|AKT|AMPK|MTOR",pathway ))
Pathway="WP_PI3KAKT_SIGNALING_PATHWAY" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)

```
\newpage

```{r}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)
```
\newpage

### REACTOME_PI3K_AKT_SIGNALING_IN_CANCER
```{r,fig.height=4,fig.width=6}
Pathway="REACTOME_PI3K_AKT_SIGNALING_IN_CANCER" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)

```
\newpage

```{r}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)
```

\newpage

### WP_MAPK_SIGNALING_PATHWAY


```{r,fig.height=6,fig.width=6,include=F}
Pathway="WP_MAPK_SIGNALING_PATHWAY" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)

```
\newpage

```{r,include=F}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)
```

### HALLMARK_MTORC1_SIGNALING
```{r,fig.height=6,fig.width=6,include=F}
Pathway="HALLMARK_MTORC1_SIGNALING" 
Stat_method="wilcox.test"
gene_list2=GSEA_pathway_list[[Pathway]]%>%as.data.frame() %>%setNames(Pathway)
Output=Singscore_data_plot_2(gene_list2,Expression_vst_MPNST_PN_q005,Expression_vst_anno_MPNST_PN_q005,GTAC_Sample_anno,Stat_method)

```


\newpage

```{r,include=F}
Output[[1]]
Valcano_plot_rnaseq (MPNST_vs_PN_res_2,GSEA_pathway_list,Pathway)
```

```{r}
knitr::knit_exit()
```


